mozilla_django_oidc_db/setup_configuration/models.py

```diff
@@ -0,0 +1,92 @@
+from typing import Optional, Union
+
+from django_setup_configuration.fields import DjangoModelRef
+from django_setup_configuration.models import ConfigurationModel
+from pydantic import AnyUrl, Discriminator, Tag
+from typing_extensions import Annotated
+
+from mozilla_django_oidc_db.models import OpenIDConnectConfig
+
+
+class OIDCFullEndpointConfig(ConfigurationModel):
+    oidc_op_authorization_endpoint: AnyUrl = DjangoModelRef(
+        OpenIDConnectConfig, "oidc_op_authorization_endpoint"
+    )
+    oidc_op_token_endpoint: AnyUrl = DjangoModelRef(
+        OpenIDConnectConfig, "oidc_op_token_endpoint"
+    )
+    oidc_op_user_endpoint: AnyUrl = DjangoModelRef(
+        OpenIDConnectConfig, "oidc_op_user_endpoint"
+    )
+
+
+class OIDCDiscoveryEndpoint(ConfigurationModel):
+    oidc_op_discovery_endpoint: AnyUrl = DjangoModelRef(
+        OpenIDConnectConfig, "oidc_op_discovery_endpoint", default=None
+    )
+
+
+def get_endpoint_endpoint_model(endpoint_data):
+
+    if isinstance(endpoint_data, dict):
+        discovery_endpoint = endpoint_data.get("oidc_op_discovery_endpoint")
```

Comment:
If discovery is provided, it will assume only discovery is wanted. If the other endpoints are provided, it will cause a model validation error.

Not sure if this is wanted, maybe if discovery fails, you might want a fallback?